Add 0-argument anonymous function example to manual/functions (#32915)
JuliaLang/julia: doc/src/manual/functions.md

as arguments. A classic example is [`map`](@ref), which applies a function to each value of
an array and returns a new array containing the resulting values:

> ```jldoctes

Reviewer: ```suggestion
```jldoctest
```
I assume this was a mistake?

Author: Yes I noticed it after submitting.

Reviewer 2: This is not resolved.

may seem strange, but is useful for "delaying" a computation. In this usage, a block of code is
wrapped in a zero-argument function, which is later invoked by calling it as `f`.

> A use case example is [`get`](@ref). 

Reviewer: ```suggestion
As an example, consider this call to [`get`](@ref):
```

```julia
get(dict, key) do
    # default value calculated here
    time()
end
```

> The code above is equivalent to calling get with an anonymous function containing the code

Reviewer: ```suggestion
The code above is equivalent to calling `get` with an anonymous function containing the code
```

> inclosed between do and end, like so:

Reviewer: ```suggestion
enclosed between `do` and `end`, like so:
```

```julia

> get( () -> time(), dict, key)

Reviewer: ```suggestion
get(()->time(), dict, key)
```

```

> The call to [`time`](@ref) is delayed by wrapping it in an argumentless anonymous function

Reviewer: ```suggestion
The call to [`time`](@ref) is delayed by wrapping it in a 0-argument anonymous function
```

> that is called only when the requested key is absent from dict. 

Reviewer: ```suggestion
that is called only when the requested key is absent from `dict`.
```
